sdk/Sdk/Tasks/ZipDeploy/ZipDeploymentStatus.cs

```diff
@@ -54,10 +53,16 @@ public async Task<DeployStatus> PollDeploymentStatusAsync(string deploymentUrl,
             {
```

Comment:
For OneDeploy /publish endpoint on flex consumption, we have the following deployment [statuses](https://msazure.visualstudio.com/Antares/_git/Antares-Kudu-Legion?path=%2Fsrc%2FKudu.Legion.Core%2FContracts%2FDeployStatus.cs&_a=contents&version=GBmain): 

Cancelled = -1 => treat as terminating with CANCELLED status *
Pending = 0 
Building = 1 
Deploying = 2 
Failed = 3 => treat as terminating with FAILED status
Success = 4 => treat as terminating with SUCCESS status
Conflict = 5 => treat as terminating with FAILED status *
PartialSuccess = 6 => treat as terminating with SUCCESS status *

* these are the new statuses to look for onedeploy on flex.

Reply:
@patelchandni - can you also confirm, while Flex is the first Functions hosting plan to support OneDeploy, we can also bring support for other hosting plans in the future right? It should not be tightly coupled with Flex?

Reply:
@patelchandni I added the rest of the `enum` values: `PartiallySuccess` and `Conflict` and modified the code to treat them as `SUCCESS` and `FAILED` respectively.

As we discussed, `Cancelled` will be treated as `Unknown` because...
 - They both have `-1` value 
 - The final result of `ZipDeploy` task is a _binary_ response: either it succeeds or not. So a 3rd response type (cancelled) cannot be added to the existing functionality.

Reply:
Thank you @anvillan

@nzthiago - Onedeploy API works quite different for other SKUs and takes different parameters. So yes, we can bring support for onedeploy for other SKUs, but the current changes are not sufficient for that. Also, some backend changes will be needed to support onedeploy on other function app SKUs.

That was one of the main reasons why I suggested to have a separate OneDeploy task so that we can put changes for all SKUs in one place in future. So if we do decide to support onedeploy for other hosting plans, then we should definitely have a separate task for onedeploy. And move flex code path to the same task.